python/graphstorm/inference/emb_infer.py

```diff
@@ -23,45 +23,53 @@
 
 
 class GSgnnEmbGenInferer(GSInferrer):
-    """ Embedding Generation inferrer.
+    """ Inferrer for embedding generation tasks.
 
-    This is a high-level inferrer wrapper that can be used directly
-    to generate embedding for inferer.
+    ``GSgnnEmbGenInferer`` inherits from the ``GSInferrer`` and use the functions
+    provided by ``GSInferrer`` to define the ``infer()`` method that performs one work:
+
+    * Generate node embeddings and save to disk.
 
     Parameters
     ----------
-    model : GSgnnNodeModel
-        The GNN model with different task.
+    model : GSgnnModel
+        This model could be one of the built-in GraphStorm prediction models, i.e.,
+        ``GSgnnNodeModel``, ``GSgnnEdgeModel``, ``GSgnnLinkPredictionModel``, or model
+        classes that inherit them and ``GSgnnNodeModelBase``, ``GSgnnEdgeModelBase``,
+        ``GSgnnLinkPredictionModelBase``.
```

Comment:
Also has a bit too many implementation details. Ideally we have one interface that this class expects, if that's `GSgnnModel`, we can say "concrete implementations of the GSgnnModel base class, for node, edge or link prediction tasks. See <here we could list GS first-party models>

Reply:
Agree with theo.

Reply:
Because GraphStorm has a different implementation of GSgnnModel from the APIs (``GSgnnNodeModelBase``, ``GSgnnEdgeModelBase``, ``GSgnnLinkPredictionModelBase``) we ask users to inherit, we have to give this detailed explanation.

If only give a brief definition, users may not know what kind of concrete implementation of `GSgnModel` is needed here.

I would prefer to give a clear instruction for users to follow although this is a little bit wordy.

Reply:
Is this an artifact of the multiple inheritance design? 

Ideally to develop a new model a user would need to create a concrete implementation of one base class, or one that inherits from one of the built-in models (node, edge, or LP) and "overrides" any functions it needs too.

My main gripe is with the sentence

> or model classes that inherit them and ``GSgnnNodeModelBase``, ``GSgnnEdgeModelBase``, ``GSgnnLinkPredictionModelBase``.

Here we are definitely pushing complexity onto the users.

Reply:
Revise to just leave the three base model classes as valid input argument values.

Reply:
Final update with help from ChatGPT:

```
       This model should be a model class that inerits ``GSgnnModel``. It is suggested
        to inherit from ``GSgnnNodeModelBase``, ``GSgnnEdgeModelBase``, or
        ``GSgnnLinkPredictionModelBase`` for node, edge, and link prediction models,
        repspectively. These bases define the necessary interfaces for each task type.
```